src/common/entityAttributes.ts

```diff
@@ -105,3 +124,67 @@ export const getFilterAttributes = (
     customKeys,
   };
 };
+
+/**
+ * Get sort attributes derived from entity attributes. Validate default sorting parameters.
+ */
+export const getSortOptions = ({
+  entityAttributes,
+  defaultSorting,
+  customColumns,
+}: {
+  /**
+   * Entity attributes for given entity.
+   */
+  entityAttributes: IEntityAttributes;
+
+  /**
+   * Optional default sorting, otherwise no sorting is applied.
+   */
+  defaultSorting?: IDefaultSorting;
+
+  /**
+   * Entity attributes can contain 5 sort attributes, but only 3 of them can be displayed.
+   */
+  customColumns?: string[];
+}): ISortOptions => {
+  const sortAttributes: ISortAttributes = {};
+
+  // Process entity attributes
+  let i = 0;
+  (customColumns ? customColumns : Object.keys(entityAttributes)).forEach((entityAttributeKey) => {
+    const entityAttribute = entityAttributes[entityAttributeKey];
+
+    // Get only entity attributes with sort property defined
+    if (entityAttribute.sort) {
+      sortAttributes[entityAttributeKey] = JSON.parse(JSON.stringify(entityAttribute)) as TSortAttribute;
+
+      // Add auto-generated table column unique index required by PatternFly library
+      if (!sortAttributes[entityAttributeKey].sort.tableColumnIndex) {
+        sortAttributes[entityAttributeKey].sort.tableColumnIndex = i;
+      } else {
+        uiLogger.error(
+          `Sort attribute 'tableColumnIndex' for key ${entityAttributeKey} is already defined, original value: ${sortAttributes[entityAttributeKey].sort.tableColumnIndex}, new value: ${i}`
+        );
+      }
+
+      i++;
+    }
+  });
```

Comment:
For now I'd like to keep it there as it's overwritten automatically, there is always a danger that property with the same name can already exists.

Reply:
Okay, then this code will do the check:

```ts
const entityAttributesEntries = customColumns
    ? Object.entries(entityAttributes).filter(([key]) => customColumns?.includes(key))
    : Object.entries(entityAttributes);

entityAttributesEntries
    .filter(([_, entityAttribute]) => entityAttribute.sort)
    .forEach(([key, entityAttribute], index) => {
      sortAttributes[key] = JSON.parse(JSON.stringify(entityAttribute)) as TSortAttribute;
   
     if (!sortAttributes[key].sort.tableColumnIndex) {
         sortAttributes[key].sort.tableColumnIndex = index;
     } else {
        uiLogger.error(
          `Sort attribute 'tableColumnIndex' for key ${key} is already defined, original value: ${sortAttributes[key].sort.tableColumnIndex}, new value: ${index}`
        );
     }
    });
 ```

Reply:
Hi @patrikk0123 , thanks for your feedback, I mention a few points why I'd keep the original solution.

Your solution means 3 cycles:
1) filter
2) filter
3) forEach

While I am creating one `forEach` and filtering `sort` and checking `tableColumnIndex` inside single cycle.

Except that if `customColumns` will contains for example only one column (this will be for example case of Build History panel), then original solution means only for cycle with 1 step + 1 condition. Your solution means
1) filter cycle going through **all** entity parameters, which does not make sense
2) starting filter cycle with 1 step
3) starting forEach cycle with 1 step

I also consider readability and maintainability of the original solution to be good.